src/compiler/crystal/config.cr

```diff
@@ -14,11 +14,13 @@ module Crystal
 
     def self.description
       version, sha = version_and_sha
-      if sha
```

Comment:
Nobody should be parsing `--version`, it's purely for humans. As you can see, the message is composed from information which is available programatically from elsewhere in this class. I don't see the complaint.